sandbox/KalturaMediaPlugin/src/com/wowza/kaltura/KalturaMediaModule.java

```diff
@@ -0,0 +1,284 @@
+import com.wowza.wms.stream.IMediaStreamActionNotify;
+import com.wowza.wms.stream.IMediaStreamActionNotify3;
+
+public class KalturaMediaModule extends ModuleBase  implements IMediaStreamActionNotify3
+{
+	static final String methodPublish = "publish";
+
+	static final String methodUnpublish = "unpublish";
+	
+	static final String methodAuthenticate = "authenticate";
+
+	private static class KalturaMediaServerConfig {
+
+		public KalturaMediaServerConfig(String path)
+		{
+			try
+			{
+				BufferedReader in = new BufferedReader( new FileReader(path) );
+				String inpitLine;
+				while((inpitLine = in.readLine()) != null){
+					String [] args = inpitLine.split("=");
+					switch(args[0])
+					{
+					case "SERVER_NODE_NAME":
+						hostName = nodeName = args[1];
+						break;
+					case "apphome_url":
+						kalturaAPIServerURL = args[1];
+						break;
+					}
+				}
+				in.close();
+				hostName = InetAddress.getLocalHost().getHostName();
+				Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+				while (interfaces.hasMoreElements()){
+				    NetworkInterface current = interfaces.nextElement();
+				    System.out.println(current);
+				    if (!current.isUp() || current.isLoopback() || current.isVirtual()) continue;
+				    Enumeration<InetAddress> addresses = current.getInetAddresses();
+				    while (addresses.hasMoreElements()){
+				        InetAddress current_addr = addresses.nextElement();
+				        if(!(current_addr instanceof Inet4Address))
+				        	continue;
+			//	        if(current_addr.isSiteLocalAddress())
+				//        	continue;
+				        if (current_addr.isLoopbackAddress()) 
+				        	continue;
+				        WMSLoggerFactory.getLogger(null).warn("KalturaMediaServerConfig : hostName: " + current_addr.getHostAddress());
+				        System.out.println(current_addr.getCanonicalHostName());
+				        hostName = current_addr.getHostAddress();
+				        break;
+				    }
+				}
+				
+			}
+			catch(Exception e)
+			{
+
+			}
+		}
+
+		private static final KalturaMediaServerConfig s_Instance = new KalturaMediaServerConfig("/opt/kaltura/app/configurations/ecdn.ini");
+
+		static public KalturaMediaServerConfig getInstance(){
+			return s_Instance;
+		}
+
+		public String nodeName;
+
+		public String hostName;
+
+		public String kalturaAPIServerURL;
+	};
+
+	static boolean reportKalturaEvent(String method,Hashtable<String,String> props)
+	{
+		return reportKalturaEvent(method,props,false);
+	}
+	
+	static String extractStreamName(IMediaStream stream)
+	{
+		WMSProperties props = stream.getClient().getProperties();
+		String [] arr = ((String)props.getProperty("connectapp")).split("/");
+		return stream.getName() + arr[arr.length - 1];
+	}
+
+	static boolean reportKalturaEvent(String method,Hashtable<String,String> props,boolean bWaitResponce)
+	{
+		WMSLoggerFactory.getLogger(null).debug(String.format("reportKalturaEvent: %s [%s] %s ",
```

Comment:
I can't see why you implement the client yourself instead of using the existing one.